src/main/java/org/dependencytrack/policy/cel/CelPolicyScriptHost.java

```diff
@@ -169,10 +172,13 @@ private static MultiValuedMap<Type, String> analyzeRequirements(final CheckedExp
                         requirements.put(TYPE_COMPONENT, "version");
                     }
                 }
+                case FUNC_COMPARE_VERSION_DISTANCE -> {
+                    requirements.putAll(TYPE_COMPONENT, List.of("purl", "uuid", "version"));
+                    requirements.put(TYPE_REPOSITORY_META_COMPONENT, "latestVersion");
+                }
                 case FUNC_COMPARE_AGE -> {
-                    if (TYPE_COMPONENT.equals(functionSignature.targetType())) {
-                        requirements.put(TYPE_COMPONENT, "published_at");
-                    }
+                    requirements.put(TYPE_COMPONENT, "purl");
+                    requirements.put(TYPE_INTEGRITY_META_COMPONENT, "publishedAt");
```

Comment:
The publish date is a field on the `Component` Protobuf type, and is named `published_at`.

```suggestion
                    requirements.put(TYPE_COMPONENT, "published_at");
```